File: yuqi/yuqi-user-web/src/main/webapp/js/controller/contentController.js
app.controller('contentController',function($scope,$http,contentService){
	
	$scope.contentList=[];//广告列表
	
	$scope.findByCategoryId=function(categoryId){
		contentService.findByCategoryId(categoryId).success(
			function(response){
				$scope.contentList[categoryId]=response;
			}
		);		
	}


	//搜索  （传递参数）
	$scope.search=function(){
		location.href="http://localhost:9104/search.html#?keywords="+$scope.keywords;
	}
        $scope.name=function(){
            $http.get("../content/name.do").success(
                function (response) {
                    $scope.loginName=response.loginName;
                }
            )
        }
    $scope.findContent=function(){
        $http.get("../content/findContent.do").success(
            function (response) {
                $scope.list=response;
            }
        )
    }
});

File: yuqi/yuqi-portal-web/src/main/java/com/yuqi/portal/controller/ContentController.java
package com.yuqi.portal.controller;
import java.util.HashMap;
import java.util.List;
import java.util.Map;


import com.yuqi.pojo.Recommend;
import com.yuqi.recommend.service.RecommendService;
import com.yuqi.sellergoods.service.ContentService;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

import com.alibaba.dubbo.config.annotation.Reference;
import com.yuqi.pojo.TbContent;



@RestController
@RequestMapping("/content")
public class ContentController {

	@Reference
	private ContentService contentService;
	@RequestMapping("/findByCategoryId")
	public List<TbContent> findByCategoryId(Long categoryId) {
		return contentService.findByCategoryId(categoryId);
	}

	@RequestMapping("/name")
	public Map findName() {
		String name = SecurityContextHolder.getContext().getAuthentication().getName();
		System.out.println("当前登录人：" + name);
		Map map = new HashMap();
		map.put("loginName", name);
		return map;
	}
	@Reference
	private RecommendService recommendService;
	@RequestMapping("/findContent")
	public List<Recommend> findContent(){
		String name = SecurityContextHolder.getContext().getAuthentication().getName();
		return  recommendService.findAll(name);
	}
}
